### ISSUE TITLE
state-dependent ncvisual detwinkling

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#1360]
#1354 didn't work out, and it was a heartbreaker. I cannot live with this bullshit twinkling; it mocks me. But in my dreams I have seen a true solution. We will add a new `ncvisual` flag, `NCVISUAL_OPTION_SMOOTH` (because this will introduce still more loss, our current solution being an optimal minimization, we must make it an opt-in). In the presence of SMOOTH, the ncvisual renders to the target plane, then takes a snapshot of it with `ncvisual_from_plane()`. This snapshot can then be fed to the blitters as state. If the visual takes a new scale, ditch the state. Otherwise, compare new cells to their previous cells. Under some fitness function, retain the previous cell or aspects thereof. Otherwise, update the state and plane as applicable.

Fitness function wise, I'm thinking idly that if we generate the same glyph, we can always update using any new coloring. Otherwise, I'm thinking if the result colors have changed drastically more than *both* the source colors (this would of course require keeping around the source colors) *and* the total RGB distance, we retain the existing solution.

This has some nice properties:
 * the first render (and only, for single-frame media) is unaffected. we only care about the goddamn twinkling, which is fundamentally a multiframe deal
 *  Basically we just dampen the sudden level sets that lead to a tremendous change in the rendered glyph, one non-representative of the changes in source.

How do we formalize "drastically more"? I'm thinking "if the source change would be lost in rasterization noise, it is not worth a major change in rasterization." Need to firm that up. But I think this is very promising.

https://www.youtube.com/watch?v=lkFMJ4-ai1I   tally-ho!

**Comment by dankamongmen** [comment]
This ought be able to key off of the TAM introduced for fast sprixel support.
